### PR TITLE
Make error message and service event message more clear

### PR DESCRIPTION
**Pull request by feiskyer**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

/kind bug
/sig cloud-provider
/area provider/azure
/priority important-soon

**What this PR does / why we need it**:

Azure cloud provider sometimes reports the following error, which is not clear:

```
processHTTPRetryResponse: backoff failure, will retry, err=Code="" Message=""
```

In such cases, we should also log the HTTP response code, so that we could learn what's the real response from ARM API.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Make error message and service event message more clear
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Review by aramase — COMMENTED** [review]
sgtm

File: staging/src/k8s.io/legacy-cloud-providers/azure/azure_backoff.go [hunk 1]
```diff
@@ -747,13 +747,15 @@ func (az *Cloud) processHTTPRetryResponse(service *v1.Service, reason string, re
 	}
 
 	if shouldRetryHTTPRequest(resp, err) {
+		message := "processHTTPRetryResponse: backoff failure, will retry"
+		if resp != nil {
+			message = fmt.Sprintf("%s, HTTP response: %d", message, resp.StatusCode)
+		}
 		if err != nil {
```
Comment (andyzhangx) [inline staging/src/k8s.io/legacy-cloud-providers/azure/azure_backoff.go]: if `resp != nil` && `err != nil`, then we only get second error msg, is that what you want? @feiskyer

Comment (feiskyer) [reply staging/src/k8s.io/legacy-cloud-providers/azure/azure_backoff.go]: it would include both

**Review by andyzhangx — APPROVED** [review]
/lgtm
/approve

**Pull request merged**